naresh/src/main/java/com/lftechnology/phpjava/ems/utlis/PasswordHashGenerator.java

```diff
@@ -0,0 +1,54 @@
+package com.lftechnology.phpjava.ems.utlis;
+
+
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+import java.security.SecureRandom;
+import java.util.Base64;
+
+public class PasswordHashGenerator {
+    // The higher the number of iterations the more
+    // expensive computing the hash is for us and
+    // also for an attacker.
+    private static final int iterations = 20 * 1000;
+    private static final int saltLen = 32;
+    private static final int desiredKeyLen = 256;
+
+    /**
+     * Computes a salted PBKDF2 hash of given plaintext password
+     * suitable for storing in a database.
+     * Empty passwords are not supported.
+     */
+    public static String getSaltedHash(String password) throws Exception {
+        byte[] salt = SecureRandom.getInstance("SHA1PRNG").generateSeed(saltLen);
+        // store the salt with the password
+        return Base64.getUrlEncoder().withoutPadding().encodeToString(salt) + "$" + hash(password, salt);
+    }
+
+    /**
+     * Checks whether given plaintext password corresponds
+     * to a stored salted hash of the password.
+     */
+    public static boolean check(String password, String stored) throws Exception {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.java.lftechnology.com:9000/coding_rules#rule_key=squid%3AS00112)